lib/api/route/streams.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:json_annotation/json_annotation.dart';
+
```

Comment:
I see, but I believe it would be more aligned with the code base to have the topics' functionality nested within streams. for example:


```dart
// lib/model/stream.dart

  /// Whether this topic should appear when not specifically focusing
  /// on this stream.
  ///
  /// This takes into account the user's visibility policy for the stream
  /// overall, as well as their policy for this topic.
  ///
  /// For UI contexts that are specific to a particular stream, see
  /// [isTopicVisibleInStream].
  bool isTopicVisible(int streamId, String topic) {
    switch (topicVisibilityPolicy(streamId, topic)) {
      case UserTopicVisibilityPolicy.none:
        switch (subscriptions[streamId]?.isMuted) {
          case false: return true;
          case true:  return false;
          case null:  return false; // not subscribed; treat like muted
        }
      case UserTopicVisibilityPolicy.muted:
        return false;
      case UserTopicVisibilityPolicy.unmuted:
      case UserTopicVisibilityPolicy.followed:
        return true;
      case UserTopicVisibilityPolicy.unknown:
        assert(false);
        return true;
    }
  }
}
```

Reply:
Also topic it self is absolutely coupled with a stream and I can't think of any functionality for a topic without having to deal with it's stream.